Add docs about setting permissions (#168)
scalar-labs/scalar-terraform: docs/CloudPermissions.md

> # Setting Cloud Permissions for scalar-terraform

Reviewer: ```suggestion
# Restricting Cloud Privileges for scalar-terraform
```

Reviewer: Permission is also fine but I think we should use permission or privilege consistently.

Author: Should the filename also be changed from `CloudPermissions.md` to `CloudPrivileges.md`?

Reviewer: Yes, that would be better !

This guide explains the privileges for users to run scalar-terraform to deploy resources to the cloud.

> In general, users who have policies or roles with higher enough privileges, such as the `*FullAccess` actions on AWS or the `Contributor` role on Azure, can deploy resources with scalar-terraform without problems. But for more secure use, you can create a dedicated user and give it limited privileges.

Reviewer: ```suggestion
In general from security perspective, it is better to assign restricted privileges to users not to make them able to do operations that they are not supposed to do.
```

## AWS

### General Policy for Deployment with scalar-terraform

> The following JSON is a general policy that allows users to manage resources for Scalar DLT with scalar-terraform.

Reviewer: Something like this ?
```suggestion
The following JSON is a policy that allows users to manage resources for Scalar DLT with scalar-terraform. Please note that it is sufficient but not necessary since, for example, it doesn't limit the target resource with `"Resource": "*"` so it can/should be further restricted.
```

> The following JSON is a general policy that allows users to manage resources for Scalar DLT with scalar-terraform.

Reviewer: Just a question. What do you mean by `general` here ? I'm just wondering if we need it.

Author: The policy lists a lot of permissions and doesn't limit the target resource (`"Resource": "*"`), so it gives users a very high privilege. There might be cases where you need to define more controlled and granular policies.

Reviewer: Ok, so maybe you mean, it is sufficient but not necessary ?
Then, for the title we can say something like `A Policy for ...` and leave a note about it as suggested below.

Author: Yes, so I will update the title and add the note as suggested. Thanks!

                "iam:AddRoleToInstanceProfile",
                "iam:AttachRolePolicy",
                "iam:CreateInstanceProfile",
                "iam:CreatePolicy",
                "iam:CreateRole",
                "iam:DeleteInstanceProfile",
                "iam:DeletePolicy",
                "iam:DeleteRole",
                "iam:DetachRolePolicy",
                "iam:GetInstanceProfile",
                "iam:GetPolicy",
                "iam:GetPolicyVersion",
                "iam:GetRole",
                "iam:ListAttachedRolePolicies",
                "iam:ListInstanceProfilesForRole",
                "iam:ListPolicyVersions",
                "iam:PassRole",
                "iam:RemoveRoleFromInstanceProfile",
                "iam:TagRole",
                "route53:*HostedZone*",
                "route53:ChangeResourceRecordSets",
                "route53:ChangeTagsForResource",
                "route53:Get*",
                "route53:List*",
                "s3:GetBucketLocation",
                "s3:GetObject",
                "s3:ListBucket"
            ],
            "Resource": "*"
        }
    ]
}
```

### Using S3 as a Terraform Backend

> If you use S3 as a Terraform backend, the action `s3:PutObject` is needed additionally.

Reviewer: ```suggestion
If you use S3 as a Terraform backend, the action `s3:PutObject` needs to be added to the above list.
```

```json
{
    "properties": {
        "roleName": "Scalar Terraform Runner",
        "description": "",
        "assignableScopes": [
            "/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
        ],
        "permissions": [
            {
                "actions": [
                    "Microsoft.Authorization/roleAssignments/*",
                    "Microsoft.Compute/availabilitySets/*",
                    "Microsoft.Compute/disks/*",
                    "Microsoft.Compute/virtualMachines/*",
                    "Microsoft.Network/loadBalancers/*",
                    "Microsoft.Network/networkInterfaces/*",
                    "Microsoft.Network/networkSecurityGroups/*",
                    "Microsoft.Network/privateDnsZones/*",
                    "Microsoft.Network/publicIPAddresses/*",
                    "Microsoft.Network/virtualNetworks/*",
                    "Microsoft.Resources/subscriptions/resourceGroups/*"
                ],
                "notActions": [],
                "dataActions": [],
                "notDataActions": []
            }
        ]
    }
}
```

### Using Azure Storage as a Terraform Backend

> If you use Azure Storage as a Terraform backend, the built-in role `Storage Account Contributor` will work.

Reviewer: ```suggestion
If you use Azure Storage as a Terraform backend, the built-in role `Storage Account Contributor` needs to be assigned additionally.
```
